### PR TITLE
MISUV-4157 Create a new connector

### PR DESCRIPTION
**Pull request by andriikondiuk**
### Description
Create a new CalculationDetailsConnector to call 1885 API. And rename the existing  CalculationDetailsConnector to CalculationDetailsConnectorLegacy for 1523

Add a link to the relevant story in Jira
[Create a new connector MISUV-4157](https://jira.tools.tax.service.gov.uk/browse/MISUV-4157)

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [ ]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [ ]  Have you checked to ensure all dependencies are up to date?
- [ ]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [x]  Have you run the journey tests?
- [x]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [x]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [ ]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [x]  Have you run the journey tests? (where applicable)
- [x]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of main?
- [x]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [x]  Have you checked the PR Builder passes?

### CODE REVIEW

File: conf/application.conf [hunk 1]
```diff
@@ -141,6 +141,7 @@ microservice {
         port = 9084
         authorisation-token {
             1523 = "secret"
+            1885 = "secret"
```
Comment (avelx) [inline conf/application.conf:144]: is that something need to be done from our side in the specific env config?

Comment (andriikondiuk) [reply conf/application.conf:144]: Yes, it's need to be done otherwise IT tests would fail

Comment (avelx) [reply conf/application.conf:144]: I am referring to prod and qa env config files, do we need to do anything in these config files?

File: app/connectors/CalculationDetailsConnector.scala [hunk 2]
```diff
@@ -28,13 +27,13 @@ class CalculationDetailsConnector @Inject()(httpClient: HttpClient,
                                             val appConfig: AppConfig)
                                            (implicit ec: ExecutionContext) extends IFConnector {
 
-  def getCalculationDetails(nino: String, calculationId: String)(implicit hc: HeaderCarrier): Future[CalculationDetailResponse]  = {
-    val getCalculationDetailsUrl: String = appConfig.ifBaseUrl + s"/income-tax/view/calculations/liability/$nino/$calculationId"
+  def getCalculationDetails(taxYear: String, nino: String, calculationId: String)(implicit hc: HeaderCarrier): Future[CalculationDetailResponse]  = {
```
Comment (avelx) [inline app/connectors/CalculationDetailsConnector.scala:30]: we might need to do some sort of verification around taxYear format;
API expecting next format => YY1-YY2

File: it/connectors/CalculationDetailsConnectorISpec.scala [hunk 3]
```diff
@@ -66,14 +68,14 @@ class CalculationDetailsConnectorISpec extends AnyWordSpec with WiremockSpec wit
     "handle errors" when {
       val desErrorBodyModel = DesErrorBodyModel("DES_CODE", "DES_REASON")
 
-      Seq(BAD_REQUEST, NOT_FOUND, CONFLICT, INTERNAL_SERVER_ERROR, SERVICE_UNAVAILABLE).foreach { status =>
+      Seq(BAD_REQUEST, NOT_FOUND, CONFLICT, UNPROCESSABLE_ENTITY, INTERNAL_SERVER_ERROR, SERVICE_UNAVAILABLE).foreach { status =>
```
Comment (avelx) [inline it/connectors/CalculationDetailsConnectorISpec.scala:71]: is that something we need to add to in order to handle INVALID_TAX_YEAR error?

**Review by avelx — APPROVED** [review]
approved

File: app/connectors/CalculationDetailsConnector.scala (hunk 2 above)
Comment (andriikondiuk) [reply app/connectors/CalculationDetailsConnector.scala:30]: It would be useful

**Pull request merged**